linode/manager
# feat: [M3-7121 & M3-7122] - Support VPC in Linode Add/Edit Config dialog

## Description 📝
Add support for VPCs in the Linode Add Config and Linode Edit Config dialogs, as well as within the config row in the Linode Configurations table.

## Major Changes 🔄
- `VPCPanel` component used in the Add/Edit Linode Configuration dialogs
  - Tweaks in `LinodeConfigDialog.tsx`, `InterfaceSelect.tsx`, and `VPCPanel.tsx`
- Network Interfaces column in Linode Config table now supports VPCs
  - new `InterfaceListItem` component

## How to test 🧪
With a properly-tagged account or with the VPC feature flag on:

- For both the "Add Configuration" and "Edit Configuration" dialogs:
  - The section title should be "Networking" and there should be updated helper text that reflects the VPC option
  -  There should be a "Primary Interface (Default Route)" dropdown
  - You should be able to successfully create a config with the designated interfaces
  - Field errors should be surfaced (example: uncheck the "Auto-assign a VPC IPv4 [...]" checkbox, input an invalid VPC IPv4, and observe the field error)
  - The dialog should reset when closed and re-opened

- For the "Edit Configuration" dialog:
  - Ensure the "Primary Interface" field displays the correct primary interface (the interface designated as "primary" based on the API payload should be in the eth position indicated by the "Primary Interface" field)
  - Changing the primary interface gets reflected in the payload submitted to the API, and after a successful submission, opening the dialog again for that config should reflect the changes you just made

None of the above should be viewable/possible without the proper account tags or the VPC feature flag being on.

Because the `<VPCPanel />` component was changed to accommodate the Add/Edit Config use case, ensure there were no regressions to its behavior in the Linode Create flow.
